sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -199,7 +199,7 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
         def createCartesianProduct() = {
           if (joinType.isInstanceOf[InnerLike]) {
-            Some(Seq(joins.CartesianProductExec(planLater(left), planLater(right), condition)))
+            Some(Seq(joins.CartesianProductExec(planLater(left), planLater(right), p.condition)))
```

Comment:
We need to write a comment above this line and explain what it is doing. 
// instead of using the condition extracted by ExtractEquiJoinKeys, we should use the original join condition, i.e., "p.condition".

Reply:
> We need to write a comment above this line and explain what it is doing.
> // instead of using the condition extracted by ExtractEquiJoinKeys, we should use the original join condition, i.e., "p.condition".

Raise a PR  https://github.com/apache/spark/pull/29084